core/domain/prod_validation_jobs_one_off.py

```diff
@@ -75,58 +66,153 @@ def _validate_external_id_relationships(cls, item):
+        for (field_name, (model_class, field_values)), external_models in zip(
+                multiple_models_keys_to_fetch.iteritems(),
+                fetched_model_instances):
+            cls.external_models[field_name] = (
+                model_class, zip(field_values, external_models))
+
+    @classmethod
+    def _get_validation_functions(cls):
+        """Returns the list of validation function to run."""
+        raise NotImplementedError
+
+    @classmethod
+    def validate(cls, item):
+        """Run _fetch_external_models and all _validate functions.
+
+        The order of running _validate is not guaranteed.
+
+        Args:
+            item: ndb.Model. Entity to validate.
+        """
+        cls.errors.clear()
+        cls.external_models.clear()
+        cls._fetch_external_models(item)
+
+        cls._validate_external_id_relationships(item)
+        for func in cls._get_validation_functions():
+            func(item)
+
 
 class UserSubscriptionsModelValidator(BaseModelValidator):
     """Class for validating UserSubscriptionsModels."""
 
     @classmethod
-    def _external_id_relationships(cls):
+    def _get_external_id_relationships(cls, item):
         return {
-            'activity_ids': exp_models.ExplorationModel,
-            'collection_ids': collection_models.CollectionModel,
+            'activity_ids': (exp_models.ExplorationModel, item.activity_ids),
+            'collection_ids': (
+                collection_models.CollectionModel,
+                item.collection_ids),
             'general_feedback_thread_ids': (
-                feedback_models.GeneralFeedbackThreadModel),
-            'creator_ids': user_models.UserSettingsModel,
-            'id': user_models.UserSettingsModel,
+                feedback_models.GeneralFeedbackThreadModel,
+                item.general_feedback_thread_ids),
+            'creator_ids': (user_models.UserSettingsModel, item.creator_ids),
+            'id': (user_models.UserSettingsModel, [item.id]),
         }
 
     @classmethod
-    def validate(cls, item):
-        """Validates a UserSubscriptionsModel entity.
+    def _get_validation_functions(cls):
+        return []
+
+
+class ExplorationModelValidator(BaseModelValidator):
+    """Class for validating ExplorationModel."""
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        state_id_mapping_model_ids = [
+            '%s.%d' % (item.id, version) for version in range(
+                1, item.version + 1)]
+        return {
+            'state_id_mapping_model': (
+                exp_models.StateIdMappingModel,
+                state_id_mapping_model_ids)
+        }
+
+    @classmethod
+    def _validate_state_name(cls, item):
+        """Validate that state name of StateIdMappingModel matches
+        corresponding ExplorationModel states.
 
         Args:
-            item: UserSubscriptionsModel. Entity to validate.
+            item: ndb.Model. Entity to validate.
```

Comment:
Could be more specific in the typeinfo, right? Both it and the description are a bit generic...